src/Items/ItemSlab.h

```diff
@@ -63,11 +69,28 @@ class cItemSlabHandler:
 				((ClickedBlockMeta & 0x08) != 0)
 			)
 			{
+				a_World.BroadcastSoundEffect(PlaceSound, a_BlockX + 0.5, a_BlockY + 0.5, a_BlockZ + 0.5, Volume, Pitch);
 				return a_Player.PlaceBlock(a_BlockX, a_BlockY, a_BlockZ, m_DoubleSlabBlockType, ClickedBlockMeta & 0x07);
 			}
 		}
 
+		// Checking the type of block that should be placed
+		AddFaceDirection(a_BlockX, a_BlockY, a_BlockZ, a_BlockFace);
+		BLOCKTYPE PlaceBlockType;
+		NIBBLETYPE PlaceBlockMeta;
+		a_World.GetBlockTypeMeta(a_BlockX, a_BlockY, a_BlockZ, PlaceBlockType, PlaceBlockMeta);
+		// If it's a slab combine into a doubleslab (means that clicked on side, top or bottom of a block adjacent to a half slab)
+		if (
+			(PlaceBlockType == m_ItemType) &&                         // Placing the same slab material
+			((PlaceBlockMeta & 0x07) == a_EquippedItem.m_ItemDamage)  // Placing the same slab sub-kind (and existing slab is single)
```

Comment:
This doesn't check the single-slabbed-ness at all. The `PlaceBlockType` check (above) does that, because double-slabs have a different block type